python/lsst/daf/butler/core/config.py

```diff
@@ -385,10 +494,37 @@ def _processExplicitIncludes(self):
                         found = fileName
                     else:
                         for dir in searchPaths:
-                            filePath = os.path.join(dir, fileName)
-                            if os.path.exists(filePath):
-                                found = os.path.normpath(os.path.abspath(filePath))
-                                break
+                            if isinstance(dir, str):
+                                filePath = os.path.join(dir, fileName)
+                                if os.path.exists(filePath):
+                                    found = os.path.normpath(os.path.abspath(filePath))
+                                    break
+                            elif isinstance(dir, ResourceDir):
+                                resource = dir.toResource(fileName)
+                                if resource.exists():
+                                    found = resource
+                                    break
+                            elif isinstance(dir, ButlerURI):
+                                if not dir.scheme:
+                                    filePath = os.path.join(dir.path, fileName)
+                                    if os.path.exists(filePath):
+                                        found = os.path.normpath(os.path.abspath(filePath))
+                                        break
+                                elif dir.scheme == "file":
+                                    # import private helper function
+                                    from .location import posix2os
+                                    filePath = posix2os(posixpath.join(dir.path, fileName))
+                                    if os.path.exists(filePath):
+                                        found = os.path.normpath(os.path.abspath(filePath))
+                                        break
+                                else:
+                                    # S3 is harder because we can't tell if
```

Comment:
You can use `head_object()` to determine if it is there by checking its metadata.

Reply:
Or use `s3utils.s3CheckFileExists()` I think.   None of the S3 support is tested at the moment. I think I'd need to add a s3config test that uploads some test configurations to the fake moto bucket and then tries to reconstruct the config from it.  I don't think @DinoBektesevic or @hsinfang are making use of DAF_BUTLER_CONFIG_PATH at the moment or using `!include` so I'm suggesting I defer implementation until we need it.